src/popup/components/TransactionItem.vue

```diff
@@ -68,45 +78,77 @@ import {
   formatDate,
   formatTime,
 } from '../utils';
-import { useTransaction } from '../../composables';
+import { useTransactionTx, useTransactionTokens } from '../../composables';
 import { useGetter } from '../../composables/vuex';
+import { ITokenResolved } from '../../types';
+
 import TransactionTokens from './TransactionTokenRows.vue';
-import Pending from '../../icons/animated-pending.svg?vue-component';
-import Reverted from '../../icons/refresh.svg?vue-component';
-import Warning from '../../icons/warning.svg?vue-component';
+
+import PendingIcon from '../../icons/animated-pending.svg?vue-component';
+import RevertedIcon from '../../icons/refresh.svg?vue-component';
+import WarningIcon from '../../icons/warning.svg?vue-component';
+import TransactionTagList from './TransactionTagList.vue';
 
 export default defineComponent({
   components: {
-    Pending,
-    Reverted,
-    Warning,
+    TransactionTagList,
+    PendingIcon,
+    RevertedIcon,
+    WarningIcon,
     TransactionTokens,
   },
   props: {
-    transaction: { type: Object as PropType<ITransaction>, required: true },
+    transaction: { type: Object as PropType<ITransaction>, default: null },
+    multisigTransaction: { type: Object as PropType<IActiveMultisigTx>, default: null },
+    hasConsensus: Boolean,
   },
   setup(props, { root }) {
     const getAmountFiat = useGetter('getAmountFiat');
 
+    const currentTx = computed<ITx | undefined>(
+      () => (props.multisigTransaction || props.transaction)?.tx,
+    );
+
     const {
       isDex,
-      tokens,
       isErrorTransaction,
-      labels,
-    } = useTransaction({
+      direction,
+      isAllowance,
+    } = useTransactionTx({
+      store: root.$store,
+      tx: currentTx.value,
+    });
+
+    const { tokens } = useTransactionTokens({
       store: root.$store,
-      initTransaction: props.transaction,
+      direction: direction.value,
+      isAllowance: isAllowance.value,
+      // TODO - refactor useTransactionTokens to use only tx
+      transaction: (props.multisigTransaction || props.transaction) as unknown as ITransaction,
     });
 
+    const redirectRoute = computed(() => {
+      if (props.multisigTransaction) {
+        return { name: 'multisig-proposal-details' };
+      }
+      const { hash } = props.multisigTransaction || props.transaction;
```

Comment:
There is already if that checks if this is a multisig transaction, so passing the `props.multisigTransaction` here makes no sense.